source/common/memory/utils.cc

```diff
@@ -0,0 +1,25 @@
+#include "common/memory/utils.h"
+
+#ifdef TCMALLOC
+
+#include "gperftools/malloc_extension.h"
+
+namespace Envoy {
+namespace Memory {
+
+void Utils::ReleaseFreeMemory() { MallocExtension::instance()->ReleaseFreeMemory(); }
```

Comment:
Could you not just simplify this whole file and do something like:
```
void Utils::ReleaseFreeMemory() {
#ifdef TCMALLOC
  MallocExtension::instance()->ReleaseFreeMemory(); 
#endif
}
```

Reply:
I was just following the style in the stats.cc file in this directory - https://github.com/envoyproxy/envoy/blob/master/source/common/memory/stats.cc. I'm fine with changing to your suggestion if you think that's better.

Reply:
Yeah, this might be a bit cleaner.

Reply:
Done

Reply:
I needed to add an extra ifdef around the include statement seems a bit ugly. Let me know if you  can think of a better way to do this.

Reply:
Why not just do:
```
#ifdef TCMALLOC
#include "gperftools/malloc_extension.h"
#endif
```

Then you can do what I put in the snippet above.

Reply:
Yup, that's what I currently have. I just thought it would be nicer to have 1 ifdef rather than 2 in this file, but it's not a big deal.